src/molecules/optionSelect/component.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+import styled from 'styled-components';
+
+import { IInputFrameTheme, IListTheme, InputFrame, List } from '../..';
+import { IconButton } from '../../atoms';
+import { Stack } from '../../layouts';
+import { Alignment, Direction } from '../../model';
+import { Box, IBoxTheme, ITextTheme, KibaIcon, Text } from '../../particles';
+import { useBuiltTheme } from '../../theming';
+import { themeToCss } from '../../util';
+import { HidingView } from '../../wrappers';
+import { defaultMoleculeProps, IMoleculeProps } from '../moleculeProps';
+
+interface IOption {
+  text: string;
+  itemKey: string;
+  isDisabled?: boolean;
+}
+
+export interface IOptionSelectTheme {
+  inputHandler: IInputFrameTheme;
+  optionsContainer: IBoxTheme;
+  optionList: IListTheme;
+  optionText: ITextTheme;
+}
+
+
+interface IStyledOptionsContainer {
+  theme: IBoxTheme;
+}
+
+const StyledOptionsContainer = styled.div<IStyledOptionsContainer>`
+  z-index: 999;
+  ${(props: IStyledOptionsContainer): string => themeToCss(props.theme)};
+`;
+
+interface IOptionSelectProps extends IMoleculeProps<IOptionSelectTheme> {
+  onItemClicked: (itemKey: string) => void;
+  options: IOption[];
+  selectedItemKey?: string;
+  isEnabled?:boolean;
+  width?: string;
+  closeIconId?: string;
+  openIconId?: string;
+  messageText?: string;
+  isFullWidth?: boolean;
+}
+
+
+export const OptionSelect = (props: IOptionSelectProps): React.ReactElement => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const width = props.isFullWidth ? '100%' : props.width || '200px';
+
+  const inputFrameTheme = useBuiltTheme('inputWrappers', 'lessPadded', props.theme?.inputHandler.inputWrapperTheme);
+
+  const optionsContainerTheme = useBuiltTheme('boxes', 'card', { margin: '0px', padding: '0px', 'border-top-left-radius': '0px', 'border-top-right-radius': '0px', ...props.theme?.optionsContainer });
+
+  const textTheme = useBuiltTheme('texts', 'default', { 'font-size': '0.9rem', ...props.theme?.optionText });
+
+  const listTheme = useBuiltTheme('listItems', 'lessPadded', props.theme?.optionList.listItems);
```

Comment:
Please suggest a better way of doing this. Like what I mean is you see I am creating new variants which don't make any sense.